doc: add FAQ for sending data to vespa.ai

rgerhards opened the pull request.

rgerhards commented:
@radu-gheorghe would you mind having a quick look?

The FAQ page is AI generated around your gist plus very little info. I am especially not sure on the vespa.ai details.

Note that I took the gist for a quick shot. In modern rsyslog, we can handle the json much easier, but we may want to provide the older method for folks who run older versions on enterprise distros.

I plan to update this page in any case as soon as the container is ready. I'd say that would be the preferred delivery platform, because it will be much easier and plug and play.

rgerhards commented:
Just FYI, this would be the current form template:

```
template(name="json-message" type="list" option.jsonftree="on") {
    property(outname="fields.message" name="msg" format="jsonf")
}
```

The old form, of course in rsyslog, still works and will continue to do so.

Current template doc: https://www.rsyslog.com/doc/configuration/templates.html

radu-gheorghe commented on doc/source/faq/vespa-ai-integration.rst:
```diff
@@ -0,0 +1,79 @@
+
+`Vespa.ai <https://vespa.ai>`_ is an open-source engine for real-time search, recommendation, and analytics at scale. It ingests structured documents,
+indexes them for low-latency search, and executes machine-learned ranking or aggregations directly where the data lives.
+Connecting rsyslog to Vespa.ai lets you ship log events into a streaming analytics platform that can keep indexes and features
```
I would change "logs" to "data" here and I would remove "streaming". While there's nothing wrong with sending logs to Vespa, and Vespa offers streaming functionality, I wouldn't call it a logging or a streaming platform.

But the rest is true, in the sense that Vespa can execute complicated ranking functions on top of massive amounts of data.

rgerhards replied:
yeah, you are of course right. That's also our track, but I have to admit that to handle fully generic data, some knobs need to be turned, and in order to make it really easy, some new knobs need to be added. But in general, we can extract and transform everything. Pure JSON of course works pretty well.

rgerhards commented on doc/source/faq/vespa-ai-integration.rst:
```diff
@@ -0,0 +1,79 @@
+  from :doc:`mmjsonparse <../configuration/modules/mmjsonparse>` or custom message properties).
+* Enable :doc:`impstats <../configuration/modules/impstats>` to monitor delivery throughput and retry behavior when Vespa slows
+  down or becomes temporarily unavailable.
+* Enable HTTP/2 ingestion—already supported by Vespa—to reduce per-message overhead and unlock future batching features (see
```
yup, agree. we need pipelinging. Will make it to the todo. There is lots going on with omhttp these days.

rgerhards commented:
@radu-gheorghe thx again for the review. If you still have time, I'd appreciate a fresh glimpse, but I can also merge (next update will be soon).

In essence, I changed log -> data and removed http/2 as this is confusing and users can actually little do right now.

radu-gheorghe commented on doc/source/faq/vespa-ai-integration.rst:
```diff
@@ -0,0 +1,78 @@
+
+`Vespa.ai <https://vespa.ai>`_ is an open-source engine for real-time search, recommendation, and analytics at scale. It ingests structured documents,
+indexes them for low-latency search, and executes machine-learned ranking or aggregations directly where the data lives.
+Connecting rsyslog to Vespa.ai lets you ship data54 events into a streaming analytics platform that can keep indexes and features
```
typo here: data54 -> data

rgerhards commented:
I did some git-fuu wrong, but after that I merged ;-) It's now in main.
